db/versions/0091.yml

```diff
@@ -0,0 +1,391 @@
+          now(),
+          taken_until_in,
+          taken_until_in -- visible initially same as taken_until
+        );
+      end
+
+  queue_claimed_task_get:
+    description: |
+      Get up to `count` tasks from the claimed queue
+    mode: write
+    serviceName: queue
+    args: visible_in timestamptz, count integer
+    returns: table (task_id text, run_id integer, taken_until timestamptz, pop_receipt uuid)
+    body: |-
+      begin
+        return query
+          with updated as (
+            update queue_claimed_tasks q
+            set pop_receipt = public.gen_random_uuid(),
+              visible = visible_in
+            where
+              q.task_id in (
+                select q2.task_id
+                from queue_claimed_tasks q2
+                where q2.visible <= now()
+                order by q2.taken_until
+                for update skip locked
+                limit count
+            )
+            returning q.task_id, q.run_id, q.taken_until, q.pop_receipt
+          )
+          select u.task_id, u.run_id, u.taken_until, u.pop_receipt
+          from updated as u
+          order by u.taken_until;
+      end
+
+  queue_claimed_task_delete:
+    description: |
+      Delete single claimed task from the queue.
+    mode: write
+    serviceName: queue
+    args: task_id_in text, pop_receipt_in uuid
+    returns: void
+    body: |-
+      begin
+        delete from queue_claimed_tasks
+        where task_id = task_id_in
+          and pop_receipt = pop_receipt_in;
+      end
+
+  queue_claimed_task_resolved:
+    description: |
+      Once the task gets resolved it is no longer relevant for the claim queue, since it cannot expire anymore.
+      We can safely delete given run from the claim queue.
+    mode: write
+    serviceName: queue
+    args: task_id_in text, run_id_in integer
+    returns: void
+    body: |-
+      begin
+        delete from queue_claimed_tasks
+        where task_id = task_id_in
+          and run_id = run_id_in;
+      end
+
+
+  # resolved queue
+  queue_resolved_task_put:
+    description: |
+      Track when task was resolved.
+      This is a short-lived record that is used by dependency resolver to update dependencies.
+      Notification is sent to `task_resolved` channel with the `task_id` as its payload.
+    mode: write
+    serviceName: queue
+    args: task_group_id_in text, task_id_in text, scheduler_id_in text, resolution_in text
+    returns: void
+    body: |-
+      begin
+        insert into queue_resolved_tasks (
+          task_group_id,
+          task_id,
+          scheduler_id,
+          resolution,
+          resolved,
+          visible
+        )
+        values (
+          task_group_id_in,
+          task_id_in,
+          scheduler_id_in,
+          resolution_in,
+          now(),
+          now()
+        );
+
+        -- notify listeners that task was resolved
+        EXECUTE 'NOTIFY task_resolved, ' || quote_literal(task_id_in) || ';';
```

Comment:
same for resolved. `DependencyResolver` would be able to know there is a new task resolved and fetch the details right after

Reply:
I'm not sure the taskId will be useful here -- this should just be a trigger for the DependencyResolver to stop sleeping and do its loop, rather than look for a specific taskId (and possibly miss other taskIds that were NOTIFY'd when it wasn't LISTENing). I suppose it doesn't hurt to include the taskId even if it's not used.

Reply:
Yes, agree. As dependency resolver takes all tasks and only needs to know that there are new records available.